### PR TITLE
Analytics UI, AnalyticsLenses, and refactoring

### PR DESCRIPTION
**Pull request by dangond-ptc**
I sincerely apologize for this monster of a commit, and wouldn't wish it upon my worst enemies, but at least now new lenses are easier to add and use throughout the codebase?

### CODE REVIEW

**Comment by dangond-ptc** [comment]
Gonna fix those lint errors, one sec

**Comment by dangond-ptc** [comment]
See also: https://github.com/ptcrealitylab/vuforia-spatial-remote-operator-addon/pull/89

File: src/humanPose/rebaScore.js [hunk 1]
```diff
@@ -676,148 +600,96 @@ function overallRebaCalculation(skel) {
...
+ */
+function calculateForPose(pose) {
+    const rebaData = extractRebaData(pose);
+    calculateReba(rebaData);
+    return rebaData;
 }
 
-export {
-    annotateHumanPoseRenderer,
+export default {
```
Comment (hobinjk-ptc) [inline src/humanPose/rebaScore.js:693]: export default has weird mismatches in its behavior between tooling and in-browser that make it best avoided (even though it works 99% of the time)

**Review by hobinjk-ptc — APPROVED** [review]
Tested locally and LGTM, minor comments inline. Excellent and thorough work!

File: css/humanPoseAnalyzerSettingsUi.css [hunk 2]
```diff
@@ -0,0 +1,121 @@
+#hpa-settings {
+    position: fixed;
+    top: 0;
+    left: 0;
+    z-index: 1000;
```
Comment (hobinjk-ptc) [inline css/humanPoseAnalyzerSettingsUi.css:5]: This z-index should be paired with a corresponding transform: translateZ(1000px)

File: src/humanPose/HumanPoseAnalyzerSettingsUi.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                </div>
+                <div class="hpa-settings-section">
+                    <div class="hpa-settings-section-title">Live Settings</div>
+                    <div class="hpa-settings-section-body">
+<!--                        <div class="hpa-settings-section-row hpa-settings-section-row-checkbox-container">-->
+<!--                            <div class="hpa-settings-section-row-label">Toggle Poses</div>-->
+<!--                            <input type="checkbox" class="hpa-settings-section-row-checkbox" id="hpa-settings-toggle-poses">-->
+<!--                        </div>-->
+                        <div class="hpa-settings-section-row hpa-settings-section-row-checkbox-container">
+                            <div class="hpa-settings-section-row-label">Toggle Spaghetti Lines</div>
```
Comment (hobinjk-ptc) [inline src/humanPose/HumanPoseAnalyzerSettingsUi.js:36]: This text would be clearer as "View Spaghetti Lines"

File: src/humanPose/draw.js [hunk 4]
```diff
@@ -437,245 +794,277 @@ export class HumanPoseAnalyzer {
...
-        const nearest = this.getNearestClone(timestamp, start);
-        if (!nearest) {
+        if (this.clones.all.length < 2) {
+            return;
+        }
+
+        const bestClone = this.getCloneByTimestamp(timestamp);
+        if (!bestClone) {
             this.hideLastDisplayedClone();
             this.lastDisplayedCloneIndex = -1;
```
[the remark this answers is not in the document]
Comment (dangond-ptc) [reply src/humanPose/draw.js:906]: You are a god for catching that, thanks

**Pull request merged**